### PR TITLE
Extract locale from file name when they are formated as `some.topic.qq.yml`

### PR DESCRIPTION
**Pull request by psychoslave**
Previously FastGettext would only handle file named `qq.yml`

### CODE REVIEW

**Comment by grosser** [comment]
hmm that's more custom bussiness-logic ... how about you extend the Yaml backend and override the lookup part to do what you indent ?

**Comment by psychoslave** [comment]
I think it's closer to the Rails default behavior, [from the manual](http://guides.rubyonrails.org/i18n.html): `Rails adds all .rb and .yml files from the config/locales directory to the translations load path, automatically.`

The PR aims to fix https://github.com/grosser/fast_gettext/issues/102 for more information.

**Comment by grosser** [comment]
I don't see anything about "." separation of topics

**Comment by psychoslave** [comment]
You are right, actually I see that the suggested way to manage that in the Rails manual is to use the directory hierarchy.

I would prefer this to lie in the gem rather than adding a workaround in the codebase. I don't know how common it might be or not to use this yaml file naming convention, so I don't know how much it's really custom logic.

All in all, with this solution it just change a single line of code and cover more cases without significant performance impact or any problem I'm aware of. I understand the willing to keep the code lean though. What do you think?

File: lib/fast_gettext/translation_repository/yaml.rb [hunk 1]
```diff
@@ -44,7 +44,8 @@ def find_and_store_files(options)
           # that is, we suppose it to be named `qq.yml` or `foo.qq.yml` where
           # `qq` stands for a locale name
           locale = File.basename(yaml_file, '.yml').split('.').last
-          @files[locale] = load_yaml(yaml_file, locale)
+          @files[locale] ||= {}
+          @files[locale].merge! load_yaml(yaml_file, locale)
```
Comment (grosser) [inline lib/fast_gettext/translation_repository/yaml.rb]: could turn that into a single line with `(@files[locale] ||= {}).merge!`

**Comment by grosser** [comment]
that looks good! ... can you add a test that fails without it ? :)

File: lib/fast_gettext/translation_repository/yaml.rb (hunk 1 above)
Comment (psychoslave) [reply lib/fast_gettext/translation_repository/yaml.rb]: :heavy_check_mark:

**Comment by grosser** [comment]
perfect! ... can you drop a tiny hint of this feature into the readme too ?

**Comment by psychoslave** [comment]
:heavy_check_mark:

**Comment by grosser** [comment]
1.7.0 🎉

**Comment by psychoslave** [comment]
:balloon:

**Pull request merged**